java/test/src/main/java/test/TestDriver.java

```diff
@@ -4,13 +4,28 @@
 
 package test;
 
+import java.io.PrintWriter;
+
 public class TestDriver {
   public static void main(String[] args) {
     int status = 0;
+
     try {
       String testClass = System.getProperty("test.class");
       Class<?> c = Class.forName(testClass);
       test.TestHelper helper = (test.TestHelper) c.getDeclaredConstructor().newInstance();
+
+      // Make the test fail if a thread fails with an unhandled exception.
+      Thread.setDefaultUncaughtExceptionHandler(
+          (Thread t, Throwable e) -> {
+            PrintWriter out = helper.getWriter();
+            out.println(
+                "!!!!!!!!!!!!!!!!! unhandled exception in thread " + t.getName() + ": " + e);
```

Comment:
If you want to leave these, getting rid of some will let this fit on one line
(which I think looks nicer than being broke up on 2 lines).
```suggestion
            out.println("!!!!!!!!!!! unhandled exception in thread " + t.getName() + ": " + e);
```